app/plugin_api.go

```diff
@@ -652,6 +652,10 @@ func (api *PluginAPI) GetPluginStatus(id string) (*model.PluginStatus, *model.Ap
 
 // KV Store Section
 
+func (api *PluginAPI) KVSetWithOptions(key string, value interface{}, options *model.PluginKVSetOptions) (bool, *model.AppError) {
```

Comment:
looks like we always expect `model.PluginKVSetOptions` to be there, in which case it might be better to turn this into a non-ptr aka `options *model.PluginKVSetOptions`.

I think this API might be a good candidate to apply  [The Functional Options Pattern](https://halls-of-valhalla.org/beta/articles/functional-options-pattern-in-go,54/). Thoughts?

https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis also goes over the pattern in detail, as well as a comparison between a `config` vs `options` approach.

Reply:
I agree that we should use `model.PluginKVSetOptions` instead of `*model.PluginKVSetOptions`. I updated the code.

While I like the function options pattern, I think the ship has sailed to use this in the plugin API. Actually I'm 2/5 that we should be consistent and use option structs.

Let me know what you think.